code/modules/power/power.dm

```diff
@@ -132,28 +132,85 @@
 
 	return A.powered(chan) // return power status of the area
 
-// increment the power usage stats for an area
-/obj/machinery/proc/use_power(amount, chan = power_channel)
-	amount = max(amount * machine_power_rectifier, 0) // make sure we don't use negative power
-	var/area/A = get_area(src) // make sure it's in an area
-	A?.use_power(amount, chan)
+/**
+ * Returns the available energy from the apc's cell and grid that can be used.
+ * Args:
+ * - consider_cell: Whether to count the energy from the APC's cell or not.
+ * Returns: The available energy the machine can access from the APC.
+ */
+/obj/machinery/proc/available_energy(consider_cell = TRUE)
+	var/area/home = get_area(src)
+
+	if(!home)
```

Comment:
```suggestion
	if(isnull(home))
```

Reply:
I'll commit it, but why should I explicitly null check this?